### PR TITLE
fix: algolia enhancement

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

Adding following supports :
- Conversion subTypes ([link](https://www.algolia.com/doc/rest-api/insights/#method-param-eventsubtype)). Subtype of the event. For conversion events either `addToCart`, or `purchase`. For all other events it should be omitted.
- adding `objectData`support ([link](https://www.algolia.com/doc/rest-api/insights/#method-param-objectdata-2))
- `currency` information ([link](https://www.algolia.com/doc/rest-api/insights/#method-param-currency))
- `price` information ([link](https://www.algolia.com/doc/rest-api/insights/#method-param-price))

## What is the related Linear task?

Resolves INT-1649

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3290/test-report.html

**Review by aashishmalik — DISMISSED** [review]

File: src/cdk/v2/destinations/algolia/procWorkflow.yaml [hunk 1]
```diff
@@ -24,6 +27,10 @@ steps:
       let eventTypeMap = $.eventTypeMapping(.destination.Config);
       let event = .message.event.trim().toLowerCase();
       let eventType = .message.properties.eventType ?? eventTypeMap[event];
+      let eventSubType = .message.properties.eventSubtype && eventType === 'conversion' ? 
+                .message.properties.eventSubtype in $.ALLOWED_EVENT_SUBTYPES ? 
+                  .message.properties.eventSubtype : null
+                : null;
```
Comment (Gauravudia) [inline src/cdk/v2/destinations/algolia/procWorkflow.yaml:33]:
```suggestion
      let eventSubType = .message.properties?.eventSubtype in $.ALLOWED_EVENT_SUBTYPES && eventType === 'conversion'  ? .message.properties.eventSubtype: null;
```

File: src/cdk/v2/destinations/algolia/procWorkflow.yaml [hunk 2]
```diff
@@ -24,6 +27,10 @@ steps:
       let eventTypeMap = $.eventTypeMapping(.destination.Config);
       let event = .message.event.trim().toLowerCase();
       let eventType = .message.properties.eventType ?? eventTypeMap[event];
```
Comment (Gauravudia) [inline src/cdk/v2/destinations/algolia/procWorkflow.yaml:29]:
```suggestion
      let eventType = .message.properties?.eventType ?? eventTypeMap[event];
```

Comment (Gauravudia) [reply src/cdk/v2/destinations/algolia/procWorkflow.yaml:29]: Similarly for all the places where we are accessing the `properties`

File: src/cdk/v2/destinations/algolia/procWorkflow.yaml [hunk 3]
```diff
@@ -32,12 +39,33 @@ steps:
...
+      const populateObjectData = function(singleProduct) {
+      singleProduct.({
+          "queryID" :  $.isDefinedAndNotNull(.queryID) ? String(.queryID) : null,
+          "price": $.isDefinedAndNotNull(.price) && $.isDefinedAndNotNull(^.message.properties.currency) ? String(.price) : null,
+          "quantity": $.isDefinedAndNotNull(.quantity)? Number(.quantity) : null,
+          "discount": $.isDefinedAndNotNull(.discount) ? String(.discount) : null
+      })
+      };
+      const products = ^.message.properties.products
+      products.map(function(singleProduct) {
```
Comment (koladilip) [inline src/cdk/v2/destinations/algolia/procWorkflow.yaml:65]: you don't need to use map, you can write
```
products.({
          "queryID" :  $.isDefinedAndNotNull(.queryID) ? String(.queryID) : null,
          "price": $.isDefinedAndNotNull(.price) && $.isDefinedAndNotNull(^.message.properties.currency) ? String(.price) : null,
          "quantity": $.isDefinedAndNotNull(.quantity)? Number(.quantity) : null,
          "discount": $.isDefinedAndNotNull(.discount) ? String(.discount) : null
      })[]
```

File: src/cdk/v2/destinations/algolia/procWorkflow.yaml [hunk 4]
```diff
@@ -24,6 +27,10 @@ steps:
       let eventTypeMap = $.eventTypeMapping(.destination.Config);
       let event = .message.event.trim().toLowerCase();
       let eventType = .message.properties.eventType ?? eventTypeMap[event];
+      let eventSubType = .message.properties.eventSubtype && eventType === 'conversion' ?
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply src/cdk/v2/destinations/algolia/procWorkflow.yaml:30]: changed

File: src/cdk/v2/destinations/algolia/procWorkflow.yaml (hunk 3 above)
Comment (shrouti1507) [reply src/cdk/v2/destinations/algolia/procWorkflow.yaml:62]: changed

**Review by koladilip — APPROVED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**